### DISCUSSION TITLE
A generic class representing functions that depend on rotations / orientations

### DISCUSSION THREAD

**Discussion by ralfHielscher** [mtex-toolbox/mtex#880] [Feature Requests]
Implement a generic class SO3Fun similar to S2Fun that can represent any kind of orientation dependent function. 
The ODF will be a special case of this. Other use cases are

* Taylor factor as a function of the orientation
* Schmid factor as a function of the orientation
* .... please provide more examples in the comments

This would allow to more easily investigate those relationships, e.g. by plotting them, finding min/max, etc.

* provide a possibility to keep the NFSOFT / NFFT plan to prevent precomputation
* allow to store / load C coefficents from a txt file

**Comment by Savillian** [comment]
I'm also looking to do this with elastic modulus as a function of orientation in the specimen reference frame.

**Reply by ralfHielscher** [reply]
Hi Savillian,

this functionality has been now included into MTEX 5.9. Could you please provide more feedback how exactly you want to use this feature?

Ralf.
